src/nvim/testdir/test_ruby.vim

```diff
@@ -0,0 +1,67 @@
+" Tests for ruby interface
+
+if !has('ruby')
+  finish
+end
+
+func Test_ruby_change_buffer()
+  call setline(line('$'), ['1 line 1'])
+  ruby Vim.command("normal /^1\n")
+  ruby $curbuf.line = "1 changed line 1"
+  call assert_equal('1 changed line 1', getline('$'))
+endfunc
+
+func Test_ruby_evaluate_list()
+  call setline(line('$'), ['2 line 2'])
+  ruby Vim.command("normal /^2\n")
+  let l = ["abc", "def"]
+  ruby << EOF
+  curline = $curbuf.line_number
+  l = Vim.evaluate("l");
+  $curbuf.append(curline, l.join("\n"))
+EOF
+  normal j
+  .rubydo $_ = $_.gsub(/\n/, '/')
+  call assert_equal('abc/def', getline('$'))
+endfunc
+
+func Test_ruby_evaluate_dict()
+  let d = {'a': 'foo', 'b': 123}
+  redir => l:out
+  ruby d = Vim.evaluate("d"); print d
+  redir END
+  call assert_equal(['{"a"=>"foo", "b"=>123}'], split(l:out, "\n"))
+endfunc
+
+func Test_ruby_evaluate_special_var()
+  let l = [v:true, v:false, v:null, v:none]
```

Comment:
```
Found errors in Test_ruby_evaluate_special_var():
Caught exception in Test_ruby_evaluate_special_var(): Vim(let):E121: Undefined variable: v:none @ function RunTheTest[35]..Test_ruby_evaluate_special_var, line 1
```

Reply:
`v:none` is a misfeature, it was intentionally removed. We can just omit it from the test.